Sources/ScipioKit/ProjectGenerator/BuildSettingsGenerator.swift

```diff
@@ -196,7 +197,47 @@ struct TargetBuildSettingsGenerator {
         )
     }
 
-    private func platformSettings(for target: ResolvedTarget, isSimulatorSupported: Bool) -> [String: XCConfigValue] {
+    private func buildHeaderSearchPaths(for target: ResolvedTarget) -> [String] {
+        var headerSearchPaths: [String] = ["$(inherited)"]
+        guard let targetDependencies = try? target.recursiveTargetDependencies() else {
+            return headerSearchPaths
+        }
+        for dependencyModule in [target] + targetDependencies {
+            switch dependencyModule.underlyingTarget {
+            case let systemTarget as SystemLibraryTarget:
+                headerSearchPaths.append("$(SRCROOT)/\(systemTarget.path.relative(to: sourceRootDir).pathString)")
+            case let clangTarget as ClangTarget:
+                headerSearchPaths.append("$(SRCROOT)/\(clangTarget.includeDir.relative(to: sourceRootDir).pathString)")
+            default:
+                continue
+            }
+        }
```

Comment:
[IMO] In this case, functional programming style might be better, because reader don't need tp take care that a value might be returned in the switch statement.
```swift

headerSearchPaths += ([target] + targetDependencies)
    .compactMap { dependencyModule in
        switch dependencyModule.underlyingTarget {
        case let systemTarget as SystemLibraryTarget:
            return "$(SRCROOT)/\(systemTarget.path.relative(to: sourceRootDir).pathString)"
        case let clangTarget as ClangTarget:
            return "$(SRCROOT)/\(clangTarget.includeDir.relative(to: sourceRootDir).pathString)"
        default:
            return nil
        }
    }
```